### DISCUSSION TITLE
Low Tunnel  Creation Success Rate

### DISCUSSION THREAD

**Discussion by PrivOci** [PurpleI2P/i2pd#1956] [Q&A]
![image](https://github.com/PurpleI2P/i2pd/assets/74867724/faea418d-3767-426a-8c37-2bc57e38b3d7)

I'm new to the i2p protocol and have been using i2pd as an alternative to the Java implementation.
I've been running the node for more than 9 hours, but the tunnel success rate has dropped from 45% to 15% and keeps decreasing. I've checked for network congestion, verified that there are no firewall or port restrictions, reviewed my i2pd configuration settings, and ensured that I'm using the latest version of i2pd.

What could be causing the low tunnel success rate and how can I improve it?

**Comment by Vort** [comment]
No one knows for sure why it is happening.
There were several attacks in recent months, maybe such effect is also caused by attacks.
Another possibility is that it is caused by some bug.
My node have low TCSR too right now:
![image](https://github.com/PurpleI2P/i2pd/assets/1242858/db88ba4a-9668-4479-9a96-94fa7eff1424)

> ensured that I'm using the latest version of i2pd

If you use 2.48.0, you can try to upgrade to version made from latest commit ae5239de435e1dcdff342961af9b506f60a494d4.
It have several improvements, which may make TCSR little higher (but not too much).

**Reply by PrivOci** [reply]
I'm using `i2pd` from `ppa:purplei2p/i2pd` repo. If you will update the repo with the fix will try it

**Reply by r4sas** [reply]
Check ppa i2pd-git.

**Reply by PrivOci** [reply]
Updated. Testing now.
Would be great if we could see i2pd version from the webconsole (update: we can)

**Reply by PrivOci** [reply]
Updated. After more than 1 hour and half it slowly went down to 22% and decreasing

**Reply by PrivOci** [reply]
Note: with Floodfills disabled it stays  over 30%
Not sure if it's related or not

**Reply by PrivOci** [reply]
After many hours it's 28%, better than before (with floodfills enabled)

**Comment by PrivOci** [comment]
I have two servers with relatively same hardware and config.
Got following results with prolonged usage:
w/ floodfills: around 10%
w/o: around 30%
